Purchases/Caching/RCInMemoryCachedObject.h

```diff
@@ -0,0 +1,29 @@
+//
+//  RCInMemoryCachedObject.h
+//  Purchases
+//
+//  Created by RevenueCat.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+@interface RCInMemoryCachedObject<ObjectType> : NSObject
+
+- (BOOL)isCacheStale;
+- (void)clearCacheTimestamp;
+- (void)clearCache;
+- (void)updateCacheTimestampWithDate:(NSDate *)date;
+- (void)cacheInstance:(ObjectType)instance
+                 date:(NSDate *)date;
+- (ObjectType _Nullable)cachedInstance;
+
+- (instancetype)initWithCacheDurationInSeconds:(int)cacheDurationInSeconds;
```

Comment:
there could also be another init with a default cache duration, but since there's only one object so far, it didn't feel terribly necessary